service/switchboardapp/src/test/java/com/switchboard/app/DecoderControllerTest.java

```diff
@@ -0,0 +1,112 @@
+package com.switchboard.app;
+
+import com.switchboard.app.controller.DecoderController;
+import com.switchboard.app.dao.DecoderDaoImpl;
+import com.switchboard.app.dao.DeviceDaoImpl;
+import com.switchboard.app.domain.DecoderEntity;
+import com.switchboard.app.domain.DeviceEntity;
+import com.switchboard.app.exceptions.DeviceNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+class DecoderControllerTest {
+
+    @InjectMocks
+    DecoderController decoderController;
+
+    @Mock
+    DecoderDaoImpl decoderService;
+
+    @Mock
+    DeviceDaoImpl deviceService;
+
+    //stubbed Objects
+    private DeviceEntity device1;
+    private DecoderEntity decoder1;
+
+    @BeforeEach
+    void setUp() throws Exception{
```

Comment:
Typo here, `setup`

Reply:
@FGRCL Where is the typo? setUp() is a junit method that he is overriding.

Reply:
You can name that method however you lik as long as you have your `BeforeEach` annotation. I just assumed @sebhani meant "setup" rather than "setUp" because that's what I am more used to seeing. It's really a minor though so I'll leave the final decision to @sebhani .

Reply:
For some reason I was under the assumption that setUp() was a method belonging to a jUnit class since I am more used to seeing that method always with a capital U. In fact, the setup method has nothing to do with jUnit, and it's mostly just a personal preference thing. @sebhani may your heart guide you in this decision haha

Reply:
I've changed it to setup